### PR TITLE
Unix socket proposal

### PR DESCRIPTION
**Pull request by choo-stripe**
Adding a proposal for a --unix-socket option

cc @kubernetes/sig-cli-proposals

### CODE REVIEW

**Comment by choo-stripe** [comment]
I signed the CLA, maybe it'll work if I comment?

File: contributors/design-proposals/cli/kubectl-unix-socket.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+## User Experience
+
+A new global option to kubectl will be added (--unix-socket)
+
+### Use Cases
+
+Issuing commands to any kube api server whereby unix socket is the preferred way to access it.
+
+kubectl --unix-socket ~/.myunixsocket --server https://myserver.com get pods
```
Comment (mikedanese) [inline contributors/design-proposals/cli/kubectl-unix-socket.md]: In this example, is the connection protocol HTTPS over UDS?

File: contributors/design-proposals/cli/kubectl-unix-socket.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+## Proposal
+
+
+A --unix-socket flag will be added to kubectl, much like in curl. The curl docs state:
+(HTTP) Connect through this Unix domain socket, instead of using the network.
+
+This is precisely what the kubectl unix-socket flag will be used for.
+
+## User Experience
```
Comment (mikedanese) [inline contributors/design-proposals/cli/kubectl-unix-socket.md]: I suggest we focus on the kubeconfig experience.

File: contributors/design-proposals/cli/kubectl-unix-socket.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+A new global option to kubectl will be added (--unix-socket)
```
Comment (mikedanese) [inline contributors/design-proposals/cli/kubectl-unix-socket.md]: This flag name conflicts with `kubectl proxy --unix-socket`

File: contributors/design-proposals/cli/kubectl-unix-socket.md (hunk 1 above)
Comment (choo-stripe) [reply contributors/design-proposals/cli/kubectl-unix-socket.md]: Ah - I was testing with just HTTP over UDS. Fixed!

File: contributors/design-proposals/cli/kubectl-unix-socket.md (hunk 2 above)
Comment (choo-stripe) [reply contributors/design-proposals/cli/kubectl-unix-socket.md]: Added an example - also tested on my machine

File: contributors/design-proposals/cli/kubectl-unix-socket.md (hunk 3 above)
Comment (choo-stripe) [reply contributors/design-proposals/cli/kubectl-unix-socket.md]: Updated the proposal (and implementation) to be `via-unix-socket`

**Comment by justaugustus** [comment]
Could we instead migrate this into a KEP and land it in [k/enhancements](https://github.com/kubernetes/enhancements)?

**Pull request closed without merging**